core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooSession.java

```diff
@@ -119,8 +119,19 @@ public static ZooKeeper connect(String host, int timeout, String scheme, byte[]
     boolean tryAgain = true;
     long sleepTime = 100;
     ZooKeeper zooKeeper = null;
-
-    long startTime = System.currentTimeMillis();
+    /*
+     * Originally, startTime = System.currentTimeMillis(). Changed to System.nanoTime() to more
+     * accurately compute durations because it is not based on system clock variations. The
+     * ZooKeeper method signature expects an int value for 'timeout' and performs several
+     * calculations that can result in Numeric Expression Overflow errors if large, nanosecond
+     * values are used. For ths reason, System.nanoTime() is is converted to MS units prior to being
+     * used in calculations. Although, the resolution of 'startTime' is still in the MS range, the
+     * value from which it is calculated is nanoTime. Also, the MS units are preserved in the
+     * original method code, without the need for conversion. TimeUnit.convert is not used because
+     * "conversions from fine to coarser granularities truncate, so lose precision" (Oracle Java 7
+     * API)
+     */
```

Comment:
I don't think any of this is necessary to keep in the code, since it is explaining why the code was changed, and not explaining the code that is there at the end.